test/unit/test_interface.py

```diff
@@ -102,3 +105,250 @@ def test_abstract_unpaired_data_loader():
+        moving_label=dummy_array,
+        fixed_label=np.random.random(size=(100, 100, 90)),
+        image_indices=[1],
+    )
+    assert "Sample [1]'s fixed image and label have different shapes. " in caplog.text
+
+    # test sample_image_label method
+    # for unlabeled input data
+    got = next(
+        generator.sample_image_label(
+            fixed_image=dummy_array,
+            moving_image=dummy_array,
+            moving_label=None,
+            fixed_label=None,
+            image_indices=[1],
+        )
+    )
+    expected = dict(
+        moving_image=dummy_array,
+        fixed_image=dummy_array,
+        indices=np.asarray([1] + [-1], dtype=np.float32),
+    )
+    assert all(np.array_equal(got[key], expected[key]) for key in expected.keys())
+
+    # for data with one label
+    got = next(
+        generator.sample_image_label(
+            fixed_image=dummy_array,
+            moving_image=dummy_array,
+            moving_label=dummy_array,
+            fixed_label=dummy_array,
+            image_indices=[1],
+        )
+    )
+    expected = dict(
+        moving_image=dummy_array,
+        fixed_image=dummy_array,
+        moving_label=dummy_array,
+        fixed_label=dummy_array,
+        indices=np.asarray([1] + [0], dtype=np.float32),
+    )
+    assert all(np.array_equal(got[key], expected[key]) for key in expected.keys())
+
+    # for data with multiple labels
+    dummy_labels = np.random.random(size=(100, 100, 100, 3))
+    got = generator.sample_image_label(
+        fixed_image=dummy_array,
+        moving_image=dummy_array,
+        moving_label=dummy_labels,
+        fixed_label=dummy_labels,
+        image_indices=[1],
+    )
+    for label_index in range(dummy_labels.shape[3]):
+        got_iter = next(got)
+        expected = dict(
+            moving_image=dummy_array,
+            fixed_image=dummy_array,
+            moving_label=dummy_labels[..., label_index],
+            fixed_label=dummy_labels[..., label_index],
+            indices=np.asarray([1] + [label_index], dtype=np.float32),
+        )
+        assert all(
+            np.array_equal(got_iter[key], expected[key]) for key in expected.keys()
+        )
+
+
+def test_file_loader():
+    """
+    Test the functions in FileLoader
+    """
+    # init, no error means passed
+    loader_grouped = FileLoader(
+        dir_paths=["/path/grouped_loader/"], name="grouped_loader", grouped=True
+    )
+    loader_ungrouped = FileLoader(
+        dir_paths=["/path/ungrouped_loader/"], name="ungrouped_loader", grouped=False
+    )
+
+    # init fails
+    with pytest.raises(ValueError) as err_info:
+        FileLoader(
+            dir_paths=["/path/ungrouped_loader/", "/path/ungrouped_loader/"],
+            name="ungrouped_loader",
+            grouped=False,
+        )
+    assert "dir_paths have repeated elements" in str(err_info.value)
+
+    # not implemented properties / functions
+    with pytest.raises(NotImplementedError):
```

Comment:
here you need a `with pytest.raises(NotImplementedError):` for each function

bcz the code will break for the first error, 
as said here https://deepregnet.github.io/DeepReg/#/CONTRIBUTING?id=test-example

Be careful, the assert is outside of the with pytest.raises(ValueError) as err_info: and we should not put multiple tests inside the same with as only the first error will be captured.